optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/BavetConstraintFactory.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.stream.bavet;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.optaplanner.core.api.domain.constraintweight.ConstraintConfiguration;
+import org.optaplanner.core.api.domain.constraintweight.ConstraintConfigurationProvider;
+import org.optaplanner.core.api.domain.constraintweight.ConstraintWeight;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.score.stream.Constraint;
+import org.optaplanner.core.api.score.stream.ConstraintFactory;
+import org.optaplanner.core.impl.domain.constraintweight.descriptor.ConstraintConfigurationDescriptor;
+import org.optaplanner.core.impl.domain.constraintweight.descriptor.ConstraintWeightDescriptor;
+import org.optaplanner.core.impl.domain.solution.descriptor.SolutionDescriptor;
+import org.optaplanner.core.impl.score.stream.ConstraintSession;
+import org.optaplanner.core.impl.score.stream.InnerConstraintFactory;
+
+public final class BavetConstraintFactory<Solution_> implements InnerConstraintFactory<Solution_> {
+
+    private final SolutionDescriptor<Solution_> solutionDescriptor;
+    private final List<BavetConstraint<Solution_>> constraintList = new ArrayList<>(20);
+
+    public BavetConstraintFactory(SolutionDescriptor<Solution_> solutionDescriptor) {
+        this.solutionDescriptor = solutionDescriptor;
+    }
+
+    @Override
+    public Constraint newConstraint(String constraintName) {
+        return newConstraint(null, constraintName);
+    }
+
+    @Override
+    public Constraint newConstraint(String constraintPackage, String constraintName) {
```

Comment:
This method does not seem to be covered by unit tests. As the examples do not have constraint streams turned on by default, it's not tested at all from the perspective of regression testing.

Reply:
This method is now gone due the side-effects refactor,
but the Conference Scheduling example will test the equivalent method.

Reply:
Ah yes, you're right. We need unit tests for constraint configuration in optaplanner-core...

Reply:
Put in https://issues.jboss.org/browse/PLANNER-1584, at the top.